pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1497,7 +1497,8 @@ type RelabelConfig struct {
 	SourceLabels []LabelName `json:"sourceLabels,omitempty"`
 
 	// Separator is the string between concatenated SourceLabels.
-	Separator string `json:"separator,omitempty"`
+	//+kubebuilder:default=";"
+	Separator string `json:"separator"`
```

Comment:
The "issue" with adding a default value is that the API response will change to return `separator: ";"`  if the field isn't set and it may disrupt some clients (reconcilers for instance).

Another option would be to change the field type to be a string pointer:
* if the pointer is nil, don't set the `separator` field in the generated Prometheus config (and let Prometheus use its own default).
* if not nil, pass the value as-is.

The risk with this option is for users that have `separator: ""` in their YAML manifests today as it will change the behavior for them.

```suggestion
	Separator *string `json:"separator"`
```

Reply:
Yes I agree this is a breaking change. But since this change is on par with Prometheus's behavior, I think it's worth doing.

Reply:
I thought of your suggested approach too. Will try this. I think it is less risky than the current (default `;` approach) because there might be less users setting `separator: ""` than users not setting it at all.

Reply:
Agreed, I don't think that `separator: ""` is widely used in the wild, especially knowing that as of today, it's the same as not setting it. If we follow that path, we's till need to mark it as a CHANGE in the changelog for safety.